lib/tree.js

```diff
@@ -12,40 +12,132 @@ class Tree {
+        
+        const findHelper = (current, key) => {
+            if ( current.key == key ){
+                return current.value
+            } else if ( current.key > key ){
+                return findHelper(current.left, key)
+            } else {
+                return findHelper(current.right, key)
+            }   
+        }
+
+        if ( !this.root ){
+            return null;
+        } else {
+            return findHelper(this.root, key)
+        }
     }
 
-    // Time Complexity: ?
-    // Space Complexity: ?
+    // Time Complexity: O(n)
+    // Space Complexity: O(n)
     inorder() {
-        throw new Error("This method hasn't been implemented yet!");
+        let resultsArray = []
+        const orderNodes = (node) =>{
+            if (node){
+                // if there is a left node, the search will access it
+                orderNodes(node.left);
+                // if there's nothing more on the left, the node will be added to the list
+                resultsArray.push({ key: node.key, value: node.value })
+                // and the right node will be accessed and looked at
+                orderNodes(node.right);
+            }
+            
+        }
+        orderNodes(this.root)
+        return resultsArray;
     }
 
-    // Time Complexity: ?
-    // Space Complexity: ?
+    // Time Complexity: O(n)
+    // Space Complexity: O(n)
     preorder() {
-        throw new Error("This method hasn't been implemented yet!");
+        let resultsArray = [];
+        const orderNodes = (node) => {
+          if (node) {
+            // the node will be added to the list when accessed
+            resultsArray.push({ key: node.key, value: node.value });
+            // if there is a left node, the search will access it
+            orderNodes(node.left);
+            // and the right node will be accessed and looked at
+            orderNodes(node.right);
+          }
+        };
+        orderNodes(this.root)
+        return resultsArray;
     }
 
-    // Time Complexity: ?
-    // Space Complexity: ?
+    // Time Complexity: O(n)
+    // Space Complexity: O(n)
     postorder() {
-        throw new Error("This method hasn't been implemented yet!");
+        let resultsArray = []
+        const orderNodes = (node) =>{
+            if (node) {
+              // if there is a left node, the search will access it
+              orderNodes(node.left);
+              // and the right node will be accessed and looked at
+              orderNodes(node.right);
+              // if there's nothing more on the left or right, the node will be added to the list
+              resultsArray.push({ key: node.key, value: node.value });
+            }
+            
+        }
+        orderNodes(this.root)
+        return resultsArray;
     }
 
-    // Time Complexity: ?
-    // Space Complexity: ?
+    // Time Complexity: O(n)
+    // Space Complexity: O(1)
     height() {
```

Comment:
👍 , but the space complexity is O(log n) for a balanced tree due to the recursive call stack.